Spec/RealtimeClientPresence.swift

```diff
@@ -1045,6 +1045,49 @@ class RealtimeClientPresence: QuickSpec {
 
                 }
 
+                // RTP2c
+                it("all presence messages from a SYNC must also be compared for newness in the same way as they would from a PRESENCE") {
+                    let options = AblyTests.commonAppSetup()
+
+                    var clientMembers: ARTRealtime!
+                    defer { clientMembers.dispose(); clientMembers.close() }
+                    waitUntil(timeout: testTimeout) { done in
+                        clientMembers = AblyTests.addMembersSequentiallyToChannel("foo", members: 120, options: options) {
+                            done()
+                            }.first
+                    }
+
+                    let client = AblyTests.newRealtime(options)
+                    defer { client.dispose(); client.close() }
+                    let channel = client.channels.get("foo")
+
+                    guard let transport = client.transport as? TestProxyTransport else {
+                        fail("TestProxyTransport is not set"); return
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        transport.beforeProcessingReceivedMessage = { protocolMessage in
+                            if protocolMessage.action == .Sync {
+                                let injectLeave = ARTPresenceMessage()
+                                injectLeave.action = .Leave
+                                injectLeave.clientId = "user110"
```

Comment:
I am not sure I follow this.  If the PresenceMessage has no `connectionId`, then how will it  be compared against the members added by `addMembersSequentiallyToChannel`.  Surely you need to make sure at least that the `connectionId` is the same.  Secondly, it would be good to test both scenarios i.e. messages that have effectively arrived before the SYNC (and thus discarded) and those that arrive after the SYNC (and thus used)